### PR TITLE
[Intel MKL] Suppressing auto_mixed_precision_test for non-avx512f machine (broadwell)

### PR DESCRIPTION
**Pull request by noim210**
[Intel MKL] Suppressing auto_mixed_precision_test for non-avx512f machine like broadwell.

### CODE REVIEW

File: tensorflow/python/framework/test_util.py [hunk 1]
```diff
@@ -1802,6 +1803,12 @@ def decorated(self, *args, **kwargs):
   return disable_test_impl
 
 
+# The description is just for documentation purposes.
+def disable_nonAVX512f(description):  # pylint: disable=unused-argument
```
Comment (penpornk) [inline tensorflow/python/framework/test_util.py:1807]: We would still want to run CUDA tests if CPU has CUDA but doesn't have AVX-512f. Maybe it's better to add a condition to `_maybe_skip` instead:
https://github.com/tensorflow/tensorflow/blob/00476037681d6e10f8c0bca83251551134b169f8/tensorflow/python/grappler/auto_mixed_precision_test.py#L372-L373

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm so sorry for the delay!

**Comment by gbaned** [comment]
@noim210  Can you please check @penpornk's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@noim210  Any update on this PR? Please. Thanks!

File: tensorflow/python/framework/test_util.py (hunk 1 above)
Comment (noim210) [reply tensorflow/python/framework/test_util.py:1807]: Hi Penporn, Sorry for the delay, I was in vacation. Did you mean to use _maybe_skip from tensorflow/python/grappler/auto_mixed_precision_test.py. 
or you are suggesting to add "cuda" condition here like below from _maybe_skip : 
  def _maybe_skip(self, mode):
    if mode == 'cuda' and not test.is_gpu_available(cuda_only=True):
      self.skipTest('No GPU is available')
    if mode == 'mkl' and not test_util.IsMklEnabled():
      self.skipTest('MKL is not enabled')

Comment (noim210) [reply tensorflow/python/framework/test_util.py:1807]: Hi Penporn, Sorry for the delay, I was in vacation. Did you mean to use _maybe_skip from tensorflow/python/grappler/auto_mixed_precision_test.py.
or you are suggesting to add "cuda" condition here like below from _maybe_skip :
def _maybe_skip(self, mode):
if mode == 'cuda' and not test.is_gpu_available(cuda_only=True):
self.skipTest('No GPU is available')
if mode == 'mkl' and not test_util.IsMklEnabled():
self.skipTest('MKL is not enabled')

Comment (noim210) [reply tensorflow/python/framework/test_util.py:1807]: If you would like me to add lines inside "_maybe_skip" function, I am guessing if mode is 'cuda' and cuda is available , the test should run (no matter if avx-512f cpu or not?). Is it possible for you to clarify little more?

**Comment by gbaned** [comment]
@noim210  Can you please resolve conflicts? Thanks!

File: tensorflow/python/framework/test_util.py (hunk 1 above)
Comment (penpornk) [reply tensorflow/python/framework/test_util.py:1828]: Hi Mazhar, Sorry for my late reply as well! I'm very far behind on github notifications lately.
I was thinking of the first option (use `_maybe_skip` and only modify `auto_mixed_precision_test.py`). But if you expect this to be used somewhere else as well, we can continue with `disable_nonAVX512f` (but not disable it for CUDA tests).

**Comment by penpornk** [comment]
(Changing the tag back to awaiting_response since I already replied in a comment.)

File: tensorflow/python/framework/test_util.py (hunk 1 above)
Comment (noim210) [reply tensorflow/python/framework/test_util.py]: I have changed it . Would you please look at it to check if that is the thing you wanted me to do?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the changes! Yes, those are what I meant. 
There are unrelated changes in `//tensorflow/python/BUILD` (probably because the file is out of date). Could you please sync to the latest commit? Thank you!

**Comment by noim210** [comment]
It looks like somebody refractor the code and move some parts from python/xx folder to python/util folder. That's why "`//tensorflow/python/BUILD`" file has some changes. 
The current changes what you are seeing, that are not my change, I just make sure current-master file and my branch's "`//tensorflow/python/BUILD`" file is same. That's why those changes are necessary (again to match with master).

**Comment by penpornk** [comment]
The changes you have don't make the file up-to-date, though. They are undoing the newer changes to the file. For example, https://github.com/tensorflow/tensorflow/blob/2f6535b09c258f021b7bff50266348c0d1dbfb13/tensorflow/python/BUILD#L1594
This line was introduced yesterday and is still there in the most recent commit. Your commit is deleting it. That's why I think your file is not up-to-date.

**Comment by noim210** [comment]
That's interesting. My master is saying those files need to be deleted. Anyway, thank you for sharing the file link. I will make sure my file is same as your link.

**Comment by noim210** [comment]
I fixed the problem. Actually, my master might be older or I forget to merge my master. 
Now I pull master and updated it accordingly.
Thank you for helping me fix this.

File: tensorflow/python/util/BUILD [hunk 2]
```diff
@@ -76,6 +76,7 @@ tf_python_pybind_extension(
         "//tensorflow/python:pybind11_lib",
         "//third_party/python_runtime:headers",
         "@pybind11",
+        "//tensorflow/core/platform:platform_port",
```
Comment (penpornk) [inline tensorflow/python/util/BUILD:79]: The list is sorted alphabetically. Could you please move this line to before `//tensorflow/python:pybind11_lib",`? Thank you very much!

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the quick fix! 
One last change please. Sorry I didn't comment on this earlier. I thought it was from merging.

File: tensorflow/python/util/BUILD (hunk 2 above)
Comment (noim210) [reply tensorflow/python/util/BUILD:79]: thank you for pointing out. Let me do this now.

Comment (noim210) [reply tensorflow/python/util/BUILD]: I fixed that. If you find any more required fix, please let me know :)

**Review by penpornk — APPROVED** [review]
Thank you again for the PR and for your patience!

**Pull request merged**